src/manager.cpp

```diff
@@ -127,12 +127,14 @@ void Manager::stop(bool discard)
     }
 }
 
-void Manager::dump(const Log& log, const std::string& channel_name, Logger::ContextInfo const& context_info)
+void Manager::dump(const Log& log, const std::string& channel_name, ContextInfo const& context_info)
 {
+    // Copy shared pointer in order to allow update without locking on set_global_context_info
+    auto context_info_handle(global_context_info_);
```

Comment:
why is this needed again?

Reply:
thread safe access in case concurrently used and set.
this will cause the pointer to not be deleted, and the map to remain valid until the function exit